File: problems_0-1000/412-FizzBuzz/go/Solution.go
package main

import (
	"fmt"
	"strconv"
)

func fizzBuzz(n int) []string {
	result := []string{}
	for i := 1; i <= n; i++ {
		if i%15 == 0 {
			result = append(result, "FizzBuzz")
		} else if i%3 == 0 {
			result = append(result, "Fizz")
		} else if i%5 == 0 {
			result = append(result, "Buzz")
		} else {
			result = append(result, strconv.Itoa(i))
		}
	}
	return result
}

func main() {
	fmt.Println(fizzBuzz(15))
}

File: problems_0-1000/412-FizzBuzz/java/Solution.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        List<String> result = solution.fizzBuzz(3);
        for (String s : result) {
            System.out.println(s);
        }
    }

    public List<String> fizzBuzz(int n) {
        List<String> result = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            if (i % 15 == 0) {
                result.add("FizzBuzz");
            } else if (i % 3 == 0) {
                result.add("Fizz");
            } else if (i % 5 == 0) {
                result.add("Buzz");
            } else {
                result.add(String.valueOf(i));
            }
        }
        return result;
    }
}
